packages/engine-ui/src/components/CarDetails.js

```diff
@@ -3,31 +3,164 @@ import styled from 'styled-components'
 import Elements from '../shared-elements'
 import RouteActivities from './RouteActivities'
 import NestedMenu from './layout/NestedMenu'
+import moment from 'moment'
+import Icons from '../assets/Icons'
+import { FlyToInterpolator } from 'react-map-gl'
+import { UIStateContext } from '../utils/UIStateContext'
 
 const Line = styled.div`
   border-top: 1px solid #dedede;
   margin: 1rem 0;
 `
 
-const CarDetails = ({ car }) => {
-  if (!car) return <p>Loading...</p>
+const Paragraph = styled.p`
+  margin-top: 0;
+  margin-bottom: 0.5rem;
+`
+
+const RouteTitleWrapper = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 10%;
+  align-items: baseline;
+  justify-items: flex-start;
+  width: 100%;
+  margin-bottom: 1rem;
+
+  button {
+    background: none;
+    border: none;
+    justify-self: end;
+  }
+
+  button:focus {
+    outline: none;
+  }
+`
+
+const CarDetails = ({ vehicle }) => {
+  const { dispatch } = React.useContext(UIStateContext)
+  const [showInfo, setShowInfo] = React.useState({
+    route: false,
+    bookings: false,
+    status: false,
+  })
+
+  if (!vehicle) return <p>Loading...</p>
+
+  const handleOnLinkClick = (id) => {
+    const activitie = vehicle.activities.filter(
+      (activitie) => activitie.id === id
+    )
+    dispatch({
+      type: 'viewport',
+      payload: {
+        latitude: activitie[0].address.lat,
+        longitude: activitie[0].address.lon,
```

Comment:
It would be easier to use array.find() here instead.. :) That way you don't have to dig in to the list that .filter returns.

```suggestion
    const activity = vehicle.activities.find(
      (activity) => activity.id === id
    )
    dispatch({
      type: 'viewport',
      payload: {
        latitude: activity.address.lat,
        longitude: activity.address.lon,
```